EventSubscriber/UserProfileSubscriber.php

```diff
@@ -43,35 +43,88 @@ public function loadUserPreferences(UserPreferenceEvent $event)
 
         $event->addPreference(
             (new UserPreference('target-weekly-hours', 32))
-                ->setValue(32)
+                ->setEnabled($isAdmin)
                 ->setType(IntegerType::class)
                 ->setOptions(['label' => 'Target Weekly Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
         );
 
         $event->addPreference(
             (new UserPreference('target-weekly-start', '1970-01-30'))
-                ->setValue('1970-01-30')
+                ->setEnabled($isAdmin)
                 ->setType(TextType::class)
                 ->setOptions(['label'    => 'Target Weekly Start Date' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
         );
 
         $event->addPreference(
-            (new UserPreference('yearly-fte-vacation-days', 35))
-                ->setValue(35)
+            (new UserPreference('yearly-vacation-hours', 168))
+                ->setEnabled($isAdmin)
                 ->setType(NumberType::class)
-                ->setOptions(['label' => 'Yearly Vacation Days' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
+                ->setOptions(['label' => 'Yearly Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
         );
 
         $event->addPreference(
             (new UserPreference('start-of-period-vacation-hours', 0))
-                ->setValue(0)
+                ->setEnabled($isAdmin)
                 ->setType(NumberType::class)
                 ->setOptions(['label' => 'Start of Period Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
-            );
+        );
+
+        $event->addPreference(
+            (new UserPreference('extra-vacation-hours', 0))
+                ->setEnabled($isAdmin)
+                ->setType(IntegerType::class)
+                ->setOptions(['label' => 'Extra Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
+        );
     }
 
     public function loadUserProfile(PrepareUserEvent $event)
     {
+        $isAdmin = $this->security->isGranted('ROLE_ADMIN');
+
+        if(!$isAdmin) {
+          // if user is not admin, we show them an unsubmittable fake input, so they know their vacation information
+          echo "<script>
+          async function showVacationInformation() {
+            // for some reason this script is included twice in the page, so we break if its already loaded
+            if (window.DS_VacationData) return;
```

Comment:
It could technically be added as many times as `loadUserProfile` is ran, so bailing out is good idea. The comment should be in PHP though and not in JavaScript

Reply:
I can, but then the relation of the comment to the immediate line containing `if` statement is not obvious. 

![image](https://github.com/user-attachments/assets/64067c20-d2cd-4e96-9531-1f3233e85c1b)

Should I?

Reply:
I means that the comment should not display in JavaScript at all or it should be changed. "For some reason" is vague. We know the reason.

Maybe you want:

In case load user profile is called multiple times, we want to ensure the form element is only added once?

Reply:
I actually did not know the reason. Thanks. Changedi t